go-controller/pkg/ovn/controller/services/services_controller.go

```diff
@@ -650,21 +682,49 @@ func (c *Controller) queueServiceForEndpointSlice(endpointSlice *discovery.Endpo
 		return
 	}
 
-	c.queue.Add(key)
+	if c.skipService(serviceNamespacedName.Name, serviceNamespacedName.Namespace) {
+		return
+	}
+	c.queue.Add(serviceNamespacedName.String())
 }
 
 // serviceControllerKey returns a controller key for a Service but derived from
 // an EndpointSlice.
-func ServiceControllerKey(endpointSlice *discovery.EndpointSlice) (string, error) {
+// Not UDN-aware, is used for egress services
```

Comment:
Yeah. I renamed it `GetServiceKeyFromEndpointSliceForDefaultNetwork` and I only call it from the egress service code. Once we add UDN support to egress services, we can make the function network aware. For now, since egress services don't support UDN, I didn't want to go as far as passing a `netInfo` from egress services to this util function.